src/node_report.cc

```diff
@@ -259,6 +267,39 @@ static void WriteNodeReport(Isolate* isolate,
 
   writer.json_arrayend();
 
+  writer.json_arraystart("workers");
+  if (env != nullptr) {
+    Mutex workers_mutex;
+    ConditionVariable notify;
+    std::vector<std::string> worker_infos;
+    size_t expected_results = 0;
+
+    env->ForEachWorker([&](Worker* w) {
+      expected_results += w->RequestInterrupt([&](Environment* env) {
+        std::ostringstream os;
+
+        GetNodeReport(env->isolate(),
+                      env,
+                      "Worker thread subreport",
+                      trigger,
+                      Local<String>(),
+                      os);
+
+        Mutex::ScopedLock lock(workers_mutex);
+        worker_infos.emplace_back(os.str());
+        notify.Signal(lock);
+      });
+    });
+
+    Mutex::ScopedLock lock(workers_mutex);
+    worker_infos.reserve(expected_results);
+    while (worker_infos.size() < expected_results)
+      notify.Wait(lock);
```

Comment:
does this this logic assume that `worker_info.size` will be always less than `expected_results` while the enumeration is on? there can be timing windows where previous worker has notified the main thread but did not start with the next worker, in which case this loop will exit pre-mature?

Reply:
> does this this logic assume that `worker_info.size` will be always less than `expected_results` while the enumeration is on? 

Yes.

> there can be timing windows where previous worker has notified the main thread but did not start with the next worker, in which case this loop will exit pre-mature?

I’m not sure if I’m understanding you correctly, but I don’t see any way in which this loop would end prematurely.

Reply:
ok, thanks, got it. I wrongly inferred that increment of `expected_results` happens in the target thread - in the nested loop, plus my test with 100 workers produced report for only few workers.

But looking closer, now realized that the outer loop is owned and run by the initiating thread, so it enumerates over all the threads before starting to wait, so no race conditions. 

Below comment helped in fixing my test too - I was't waiting for workers to come online, so!

thanks, it clears up my questions.